var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -158,13 +166,18 @@ class Dd4hep(CMakePackage):
     depends_on("geant4@10.2.2:", when="+ddg4")
     depends_on("assimp@5.0.2:", when="+ddcad")
     depends_on("hepmc3", when="+hepmc3")
+    depends_on("hepmc3@3.2.6:", when="+hepmc3-gz")
+    depends_on("bzip2", when="+hepmc3-gz")
+    depends_on("lzma", when="+hepmc3-gz")
+    depends_on("zlib", when="+hepmc3-gz")
```

Comment:
Should be
```suggestion
    depends_on("zlib-api", when="+hepmc3-gz")
```
since #37372

Reply:
Maybe we need `lzma-api` too as a provider for `lzma` and `xz`...